lib/puppet/functions/foreman_proxy/generate_uuid.rb

```diff
@@ -0,0 +1,9 @@
+Puppet::Functions.create_function(:'foreman_proxy::generate_uuid') do
```

Comment:
The same function implementation is in puppet-foreman and this module already depends on that. It would be better to call the function from that module to reduce duplication.

Reply:
The same applies to the function for generating the signature, it is on my todo list